Kernel/Term.cpp

```diff
@@ -37,7 +37,9 @@ constexpr unsigned Term::SPECIAL_FUNCTOR_LOWER_BOUND;
 
 void Term::setId(unsigned id)
 {
-  if (env.options->randomTraversals()) {
+  if (env.options->randomTraversals() &&
+      Random::seed() != 1) { // not until a proper seed has been set (i.e. after parsing!)
+      // (cf ProvingHelper::runVampire and getPreprocessedProblem in vampire.cpp)
     id += Random::getInteger(1 << 12) << 20; // the twelve most significant bits are randomized
```

Comment:
Also, could you remind me why this doesn't lead to collisions? 2^20 doesn't seem so large a space.

Reply:
It will lead to a collision. But what are the changes this will happen much earlier than after assigning an id to almost all the 2^32 terms (after which we get an overflow even without `randomTraversals`)?

Note that this's is `id +=`. I wouldn't mind using xor. But the 32 bits are NOT split into a random part and an incremented part.